algorithms task (6)/student/include/impl.h

```diff
@@ -19,4 +21,16 @@
  */
 
 template <typename Sequence, typename FwIt>
-void insert_to_sorted(Sequence& target, const FwIt sourceBegin, const FwIt sourceEnd);
+void insert_to_sorted(Sequence& target, const FwIt sourceBegin, const FwIt sourceEnd)
+{
+    static_assert(std::is_const<typename std::remove_reference<
+                      typename std::iterator_traits<FwIt>::reference>::type>::value,
+                  "FwIt must point to const data");
+
+    // Use std::upper_bound() for stable order of source values.
+
+    std::for_each(sourceBegin, sourceEnd, [&target](typename std::iterator_traits<FwIt>::reference sourceValue)
```

Comment:
why not const reference?

Reply:
Because we've already tested by static_assert() that reference is constant.
Anyway changed to "const value_type&" as it is clearer to understand, what's your reckon?